protocols/request-response/src/lib.rs

```diff
@@ -61,7 +61,7 @@
 pub mod codec;
 pub mod handler;
 
-pub use codec::{ProtocolName, RequestResponseCodec};
```

Comment:
https://github.com/rust-lang/rust/issues/41517 is not stable, so `RequestResponseCodec` to `Core` rename is probably breaking.

Reply:
Damn. Will this work though?

1. Create a new trait `Codec`
1. Create a blanket impl of `Codec` for everything that is `RequestReponseCodec`
1. Deprecate `RequestResponseCodec`
1. Use `Codec` everywhere in `libp2p-request-response`

Reply:
ehe nice! Updated, yeah it works at least to give a warning to users using `RequestResponseCodec` pointing how to update. But it will be still breaking cause `RequestResponse::new` expects `Codec` now

Reply:
Damn. It would be nice to make it completely non-breaking.

Reply:
> ehe nice! Updated, yeah it works at least to give a warning to users using `RequestResponseCodec` pointing how to update. But it will be still breaking cause `RequestResponse::new` expects `Codec` now

Is this actually breaking, now that we have the blanket impl where we implement `Codec` for everything that implements `RequestResponseCodec`?

Reply:
yeah good point Thomas, it probably isn't yeah AFAICT. Should we then make this a patch version?

Reply:
> yeah good point Thomas, it probably isn't yeah AFAICT. Should we then make this a patch version?

I'd much prefer so, yes!

Reply:
Ok, updated!